planning/behavior_path_planner/src/utils/goal_planner/goal_searcher.cpp

```diff
@@ -38,6 +38,33 @@ using lanelet::autoware::NoStoppingArea;
 using tier4_autoware_utils::calcOffsetPose;
 using tier4_autoware_utils::inverseTransformPose;
 
+// Sort with smaller longitudinal distances taking precedence over smaller lateral distances.
+struct SortByLongitudinalDistance
+{
+  bool operator()(const GoalCandidate & a, const GoalCandidate & b) const noexcept
+  {
+    const double diff = a.distance_from_original_goal - b.distance_from_original_goal;
+    constexpr double eps = 0.01;
+    if (std::abs(diff) < eps) {
+      return a.lateral_offset < b.lateral_offset;
+    }
+    return a.distance_from_original_goal < b.distance_from_original_goal;
+  }
+};
+
+// Sort with weighted lateral distance against longitudinal distance
```

Comment:
[NITS]
If my understanding is correct, comments below is easier to understand for me.
Sort with the weighted sum of the longitudinal distance and the lateral distance weighted by lateral_cost.

Reply:
thanks! it is better! fixed in https://github.com/autowarefoundation/autoware.universe/pull/5110/commits/e6f3068375f3246ebaa10c448e03ceb3f2dec7da